docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,261 @@
+  - Source files may be small but the generated CSS could be huge.
+
+## Pros and Cons of the Email Previewing Options
+
+It’s notoriously difficult to preview what different email states look like in FxA for Engineering, Design, and Product. Currently, we're manually previewing the templates by running `yarn write-emails` in the `fxa-auth-server` root directory which doesn't account for states unless we manually change variables and run the command again. Furthermore, even the Maildev client requires us to manually trigger the email when running FxA locally so we need a robust solution to account for these challenges faced by the team.
+
+### Option A - Continue to use the `write-emails` command
+
+- Description
+  - Continue exclusively using the `yarn write-emails` command to generate and preview email templates.
+- Pros
+  - The effort to refactor all the templates would be eliminated for now.
+- Cons
+  - With this approach, we cannot preview the various email states simultaneously and it requires us to manually change the variables and trigger the command again.
+  - The script associated with the `write-emails` command can break if we add/remove imports for `senders`
+
+### Option B - Use Storybook
+
+- Description
+  - Use [Storybook](https://storybook.js.org/) to preview the email templates
+- Pros
+  - With Storybook, we can preview all of the different email states simultaneously which was not possible before with Option A
+  - Storybook deployment has worked great so far for a few other FxA services so hooking up the client for `fxa-auth-server` to view the modernized components might result in consistency.
+- Cons
+  - We're still going to be seeing an HTML version of the email in Storybook that may look different in a web client, however, we could mitigate that by previewing it in Maildev.
+  - While Storybook is easy to setup and configure, it requires the service to be hosted somewhere in GCP and might have some maintaince requirements in future.
+
+### Option C - Use Mailtrap
+
+- Description
+  - [Mailtrap](https://mailtrap.io/) is an online tool for safe email testing in dev and staging environments. It is frequently used as a fake SMTP server to catch test emails, view them in virtual inboxes, and debug before sending to real users.
+- Pros
+  - Mailtrap provides us the functionality to view all Bcc’ed addresses and to verify whether they stay invisible for the main message recipients.
+  - It can let us view several test results at once like HTML preview, HTML and CSS validation and also outputs the line of code where errors are found.
+  - With Mailtrap's automatic and manual email forwarding, we can share our testing results with fellow team members either by sending them manually or automatically by setting a list of forwarding rules.
+- Cons
+  - Mailtrap is a paid online service which may not be a part of our requirements as of now.
```

Comment:
Have we check if Mozilla already pays for a service like Litmus or Mailtrap?

Reply:
I may not be aware of that. @LZoog would you like to add anything here?